kani-driver/src/get_det_vals.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use std::fs;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::{Path, PathBuf};
+
+use crate::session::KaniSession;
+use crate::util::alter_extension;
+use anyhow::Result;
+use kani_metadata::HarnessMetadata;
+use serde_json::Value;
+
+impl KaniSession {
```

Comment:
You should separate the parser code from cbmc execution code. The parser shouldn't be part of KaniSession. This is getting over-used.